src/Sylius/Bundle/ApiBundle/CommandHandler/Changer/CommandPaymentMethodChanger.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\CommandHandler\Changer;
+
+use Sylius\Bundle\ApiBundle\Command\AbstractPaymentMethod;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Sylius\Component\Core\Repository\PaymentMethodRepositoryInterface;
+use Sylius\Component\Core\Repository\PaymentRepositoryInterface;
+use Sylius\Component\Payment\Model\PaymentInterface;
+use Webmozart\Assert\Assert;
+
+/** @experimental */
+final class CommandPaymentMethodChanger implements CommandPaymentMethodChangerInterface
+{
+    /** @var PaymentRepositoryInterface */
+    private $paymentRepository;
+
+    /** @var PaymentMethodRepositoryInterface */
+    private $paymentMethodRepository;
+
+    public function __construct(
+        PaymentRepositoryInterface $paymentRepository,
+        PaymentMethodRepositoryInterface $paymentMethodRepository
+    ) {
+        $this->paymentRepository = $paymentRepository;
+        $this->paymentMethodRepository = $paymentMethodRepository;
+    }
+
+    public function changePaymentMethod(
+        AbstractPaymentMethod $abstractPaymentMethod,
```

Comment:
I would be for passing here `paymentId` and `paymentMethodCode` instead of abstract command